pkg/util/arm/resources.go

```diff
@@ -0,0 +1,66 @@
+package arm
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// ArmResource represents a resource and its child resources.
+// Typically we would use the autorest package for this, but
+// It does not have support for subresources
+type ArmResource struct {
+	SubscriptionID string
+	ResourceGroup  string
+	Provider       string
+	ResourceName   string
+	ResourceType   string
+	SubResource    SubResource
+}
+
+// SubResource represents an ARM Proxy Resource
+// ARM supports up to 3 levels of nested resources
+// https://eng.ms/docs/products/arm/api_contracts/guidelines/rpc#rpc030-avoid-excessive-resource-type-nesting
+type SubResource struct {
+	ResourceName string
+	ResourceType string
+	SubResource  *SubResource
+}
+
+// ParentResourcetoString returns a string of the parent object in form of azureResourceID
+func (r ArmResource) ParentResourceToString() string {
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/%s/%s/%s", r.SubscriptionID, r.ResourceGroup, r.Provider, r.ResourceType, r.ResourceName)
+}
+
+// String function returns a string in form of azureResourceID
+func (r ArmResource) String() string {
```

Comment:
technically this should see how far deep a arm resource is.  i.e. keep iterating through subresource until nil 

on line 44 TODO comment.